exporters/jaeger/src/recordable.cc

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <opentelemetry/exporters/jaeger/recordable.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace jaeger
+{
+
+Recordable::Recordable() : span_{new thrift::Span} {}
+
+void Recordable::PopulateAttribute(nostd::string_view key,
+                                   const opentelemetry::common::AttributeValue &value)
+{
+  if (nostd::holds_alternative<int64_t>(value))
+  {
+    AddTag(std::string{key}, nostd::get<int64_t>(value));
+  }
+  else if (nostd::holds_alternative<bool>(value))
+  {
+    AddTag(std::string{key}, nostd::get<bool>(value));
+  }
+  else if (nostd::holds_alternative<double>(value))
+  {
+    AddTag(std::string{key}, nostd::get<double>(value));
+  }
+  else if (nostd::holds_alternative<nostd::string_view>(value))
+  {
+    AddTag(std::string{key}, std::string{nostd::get<nostd::string_view>(value)});
+  }
+  // TODO: extend other AttributeType to the types supported by Jaeger.
+}
+
+void Recordable::SetIds(trace::TraceId trace_id,
+                        trace::SpanId span_id,
+                        trace::SpanId parent_span_id) noexcept
+{
+  span_->__set_traceIdLow(*(reinterpret_cast<const int64_t *>(trace_id.Id().data())));
+  span_->__set_traceIdHigh(*(reinterpret_cast<const int64_t *>(trace_id.Id().data()) + 1));
+  span_->__set_spanId(*(reinterpret_cast<const int64_t *>(span_id.Id().data())));
+  span_->__set_parentSpanId(*(reinterpret_cast<const int64_t *>(parent_span_id.Id().data())));
+}
+
+void Recordable::SetAttribute(nostd::string_view key,
+                              const opentelemetry::common::AttributeValue &value) noexcept
+{
+  PopulateAttribute(key, value);
+}
+
+void Recordable::AddEvent(nostd::string_view name,
+                          core::SystemTimestamp timestamp,
+                          const common::KeyValueIterable &attributes) noexcept
+{
+  // TODO: convert event to Jaeger Log
+}
+
+void Recordable::AddLink(const opentelemetry::trace::SpanContext &span_context,
+                         const common::KeyValueIterable &attributes) noexcept
+{
+  // TODO: convert link to SpanRefernece
+}
+
+void Recordable::SetStatus(trace::StatusCode code, nostd::string_view description) noexcept
+{
+  if (code == trace::StatusCode::kUnset)
+  {
+    return;
```

Comment:
shouldn't return here, as the description should still be needed to set.

Reply:
The conversion doc says the status (Jaeger span status?) should not be set for `UNSET`, let me know if this is still needed.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#status